spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestGenerateSymlinkFormatManifestsProcedure.java

```diff
@@ -0,0 +1,153 @@
+    List<Object[]> result = sql("CALL %s.system.generate_symlink_format_manifest('%s')", catalogName, tableIdent);
+    Table table = validationCatalog.loadTable(tableIdent);
+    String location = table.location();
+    List<Object[]> expected = Lists.newArrayList();
+    expected.add(row(table.currentSnapshot().snapshotId(), 6L));
+    assertEquals("Should find 6 files", expected, result);
+    checkDirectoryExists(table.location(), "_symlink_format_manifest");
+    checkDirectoryExists(table.location(), "_symlink_format_manifest",
+        Long.toString(table.currentSnapshot().snapshotId()));
+    checkDirectoryExists(table.location(), "_symlink_format_manifest",
+        Long.toString(table.currentSnapshot().snapshotId()), "data_bucket=15");
+    checkDirectoryExists(table.location(), "_symlink_format_manifest",
+        Long.toString(table.currentSnapshot().snapshotId()), "data_bucket=2");
+    checkDirectoryExists(table.location(), "_symlink_format_manifest",
+        Long.toString(table.currentSnapshot().snapshotId()), "data_bucket=3");
+  }
+
+  @Test
+  public void testGenerateSymlinkFormatManifestsCustomLocationPositionArgument() {
+    sql("CREATE TABLE %s (id bigint NOT NULL, data string) USING iceberg", tableName);
+    sql("INSERT INTO TABLE %s VALUES (1, 'a')", tableName);
+    sql("INSERT INTO TABLE %s VALUES (2, 'b')", tableName);
+    Table table = validationCatalog.loadTable(tableIdent);
+    String customLocation = table.location() + "/" + UUID.randomUUID();
+    List<Object[]> result = sql("CALL %s.system.generate_symlink_format_manifest('%s', '%s')",
+        catalogName, tableIdent, customLocation);
+    List<Object[]> expected = Lists.newArrayList();
+    expected.add(row(table.currentSnapshot().snapshotId(), 2L));
+    assertEquals("Should find 2 files", expected, result);
+    checkDirectoryExists(customLocation);
+  }
+
+  @Test
+  public void testGenerateSymlinkFormatManifestsCustomLocationNamedArgument() {
+    sql("CREATE TABLE %s (id bigint NOT NULL, data string) USING iceberg", tableName);
+    sql("INSERT INTO TABLE %s VALUES (1, 'a')", tableName);
+    sql("INSERT INTO TABLE %s VALUES (2, 'b')", tableName);
+    Table table = validationCatalog.loadTable(tableIdent);
+    String customLocation = table.location() + "/" + UUID.randomUUID();
+    List<Object[]> result = sql(
+        "CALL %s.system.generate_symlink_format_manifest(symlink_root_location => '%s', table => '%s')",
+        catalogName, customLocation, tableIdent);
+    List<Object[]> expected = Lists.newArrayList();
+    expected.add(row(table.currentSnapshot().snapshotId(), 2L));
+    assertEquals("Should find 2 files", expected, result);
+    checkDirectoryExists(customLocation);
+  }
+
+  private void checkDirectoryExists(String... paths) {
+    String path = String.join("/", paths);
+    Assert.assertTrue("Directory should exist: " + path, Files.isDirectory(Paths.get(URI.create(path))));
+  }
```

Comment:
should we also check the symlink folder when being used as the external table is return's same result as iceberg table ? WDYT